### PR TITLE
[WIP] Add Comments API

### PR DESCRIPTION
**Pull request by obsc**
Added functionality for https://developer.github.com/v3/repos/comments/

Implemented using the updated style.
Still working on expanding coverage to gist comments and issue comments.

### CODE REVIEW

File: octokit/comments.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+	success = (result.Response.StatusCode == 204)
+	return
+}
+
+type Comment struct {
+	*hypermedia.HALResource
+
+	ID        int        `json:"id,omitempty"`
+	URL       string     `json:"url,omitempty"`
+	HTMLURL   string     `json:"html_url,omitempty"`
```
Comment (pengwynn) [inline octokit/comments.go]: Looks like we've got a mix of `HTMLURL` and `HtmlURL` in the project. @jingweno got a preference?

**Comment by pengwynn** [comment]
This is looking good. :sparkles:

**Comment by obsc** [comment]
Endpoint should be complete with gist and issue comments now.

https://developer.github.com/v3/gists/comments/
https://developer.github.com/v3/issues/comments/

File: octokit/commit_comments_test.go [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+		respondWithStatus(w, 204)
+	})
+
+	success, result := client.CommitComments().Delete(nil, M{"owner": "octokit", "repo": "go-octokit", "id": 4236029})
+	assert.False(t, result.HasError())
+
+	assert.True(t, success)
+}
+
+func TestCommitCommentsService_Failure(t *testing.T) {
```
Comment (pengwynn) [inline octokit/commit_comments_test.go]: Nice. I like this test.

File: octokit/issue_comments.go [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+type IssueComment struct {
+	*hypermedia.HALResource
+
+	ID        int        `json:"id,omitempty"`
+	URL       string     `json:"url,omitempty"`
+	User      User       `json:"user,omitempty"`
+	CreatedAt *time.Time `json:"created_at,omitempty"`
+	UpdatedAt *time.Time `json:"updated_at,omitempty"`
+	Body      string     `json:"body,omitempty"`
+	HTMLURL   string     `json:"html_url,omitempty"`
```
Comment (pengwynn) [inline octokit/issue_comments.go]: Do we need the `*URL` fields here? I was thinking the embedded `HALResource` pulls all the link relations into `.Rels()` automagically.

**Comment by pengwynn** [comment]
@obsc Nice. One small question then I think we can merge this.

File: octokit/issue_comments.go (hunk 3 above)
Comment (obsc) [inline octokit/issue_comments.go]: Hmm, I'm not entirely familiar with how `HALResource` works with automatically linking relations.

Is there any particular reason that repositories.go has `URL` field? It already embeds a `HALResource` as well.
https://github.com/octokit/go-octokit/blob/master/octokit/repositories.go#L65-L66

Comment (pengwynn) [inline octokit/issue_comments.go]: @obsc I'm not sure. I thought Sawyer's [HALResource][] was autolinking those, but I'm only seeing it handle the `_links` hash. I see the [root][] handles them differently. I'd like to have those automatically parsed if we can. We don't need to hold up this PR, but would you mind digging into that?

[root]: https://github.com/octokit/go-octokit/blob/0000dde833b0f66eb38c6ba071b1845f2ff96084/octokit/root.go

[HALResource]: https://github.com/lostisland/go-sawyer/blob/master/hypermedia/hypermedia.go

**Pull request merged**